src/utils/formHelperFunctions.ts

```diff
@@ -90,6 +91,123 @@ export const formatAllFormFields = (steps: any, forUser = false) => {
   return formattedFields;
 };
 
+export const getAllFields = (fieldKeys: string[]): Record<string, Field> => {
+  const fields: Record<string, Field> = {};
+  fieldKeys.forEach((key) => {
+    const newField = new Field(key);
+    fields[key] = newField;
+  });
+  return fields;
+};
+
+export function isValidFieldIdentifier(str: string) {
```

Comment:
would it be easier to check if an identifier is present within our list of field ids rather than checking that it is *not* every possible global keyword? which seems brittle / edge-casey

Reply:
The issue is that the form designer could assign a field id that collides with a javascript reserved word.  They will get a validation error should they try to use it in a rule.  However, even if they do not use it in a rule, the runtime injects that field and this causes an exception at runtime due to the reserved word being used.  So to keep things robust we need to avoid injecting fields with reserved word ids/keys.  Yes it is an edge case but if someone were to hit it, then it would be a big time waster.  So I really think this check against reserved words needs to remain in place.

Reply:
can we add a comment explaining why we're doing this check then?

Reply:
sure